ports/esp32/main_esp32/idf_component.yml

```diff
@@ -1,5 +1,6 @@
 ## IDF Component Manager Manifest File
 dependencies:
+  espressif/lan867x: "^1.0.0"
```

Comment:
This is annoying because the component depends on idf >= 5.3.0 so the build will fail on older ESP-IDF, and AFAIK there's no notion of a conditional dependency in the IDF Component Manager Manifest format.

I think our options are:

1. Hold this PR until we only support IDF 5.3.0 or newer
2. Generate this file as part of the build. :grimacing: 
3. Add a second `main_esp32` component (i.e. `main_esp32_old`) and pick one or the other in the top-level CMakeLists.txt file depending on the ESP-IDF version.

I can have a crack at option 2 if you want to try that path.

Reply:
I suggest we go for option 3, it's the simplest way to move forward.